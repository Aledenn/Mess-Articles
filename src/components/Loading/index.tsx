import React, { SFC } from 'react';
import styled, { keyframes } from 'styled-components';
import classnames from 'classnames';

const colors = ['#7ef9ff', '#89cff0', '#4682b4', '#0f52ba', '#000080'];

const wave = keyframes`
  50%,
  75% {
    transform: scale(2.5);
  }

  80%,
  100% {
    opacity: 0;
  }
`;

const LoadingContainer = styled.div`
  display: none;
  &.show {
    display: flex;
  }
  animation-delay: 1s;
  .dot {
    position: relative;
    width: 2em;
    height: 2em;
    margin: 0.8em;
    border-radius: 50%;

    &::before {
      position: absolute;
      content: '';
      width: 100%;
      height: 100%;
      background: inherit;
      border-radius: inherit;
      animation: ${wave} 2s ease-out infinite;
    }
    ${colors.map(
      (color, index) => `
        &:nth-child(${index + 1}){
          background: ${color};
          &::before {
            animation-delay: ${(index + 1) * 0.2}s;
          }
        }
        `,
    )}
  }
`;

interface IProps {
  loading: boolean;
  className?: string;
}

const Loading: SFC<IProps> = ({ loading = false, className }) => (
  <LoadingContainer
    className={classnames(className, {
      show: loading === true,
    })}
  >
    <div className="dot" />
    <div className="dot" />
    <div className="dot" />
    <div className="dot" />
    <div className="dot" />
  </LoadingContainer>
);

export default Loading;
