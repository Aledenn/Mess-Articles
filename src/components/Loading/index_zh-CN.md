---
name: Loading
edit: false
sidebar: true
route: /Loading
---

import Demo1 from './demo/demo1'; 
import Demo2,{DemoCss} from './demo/demo2';

# Loading

一个生动的 Loading 组件

## 代码演示

### 基本使用

<Demo1/>

```
const DemoLoading = () => {
  return <Loading loading />;
};
```

### 自定义 sytle
需要配合styled-components使用

<Demo2/>

```
import Loading from '..';

const MyLoading = styled(Loading)`
  .dot {
    height: 1em;
    width: 1em;
  }
`;

const Demo2 = () => <MyLoading loading />;
```

或者直接通过className修改
<DemoCss/>

```
import './index.less';

DemoCss = () => <Loading loading className="demo_css" />

// index.less
.demo_css {
  .dot {
    height: 1em;
    width: 1em;
  }
}
```


## API

| 参数    | 说明              | 类型    |
| ------- | ----------------- | ------- |
| loading | 控制 loading 状态 | boolean |
| className | 自定义组件样式 | string |
