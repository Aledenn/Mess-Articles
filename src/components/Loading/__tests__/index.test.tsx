import React from 'react';
import { render } from '@testing-library/react';
import Loading from '..';

describe('Loading', () => {
  test('no loading', () => {
    const { container } = render(<Loading loading={false} />);
    expect(container).toMatchSnapshot();
  });

  test('loading', () => {
    const { container } = render(<Loading loading />);
    expect(container).toMatchSnapshot();
  });
});
