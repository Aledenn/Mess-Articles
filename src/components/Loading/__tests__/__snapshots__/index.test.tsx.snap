// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading className 1`] = `
<div>
  <div
    class="sc-bdVaJa jjICzf cbsama show"
  >
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
  </div>
</div>
`;

exports[`Loading loading 1`] = `
<div>
  <div
    class="sc-bdVaJa jjICzf show"
  >
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
  </div>
</div>
`;

exports[`Loading no loading 1`] = `
<div>
  <div
    class="sc-bdVaJa jjICzf"
  >
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
  </div>
</div>
`;
